agent/grpc-external/services/dns/server.go

```diff
@@ -0,0 +1,136 @@
+package dns
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/miekg/dns"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/consul/proto-public/pbdns"
+)
+
+type Local struct {
+	IP   net.IP
+	Port int
+}
+
+type Config struct {
+	Logger       hclog.Logger
+	DNSServeMux  *dns.ServeMux
+	LocalAddress Local
+}
+
+type Server struct {
+	Config
+}
```

Comment:
Is it worth us combining these structs? Not sure what benefit we get from a separate `Config`?

Reply:
I think it's simply to allow the `NewServer` method to not have to have individual arguments.